back/sssscs/src/main/java/com/ib/certificate/CertificateService.java

```diff
@@ -0,0 +1,45 @@
+package com.ib.certificate;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.ib.certificate.Certificate.Type;
+import com.ib.certificate.CertificateRequest.Status;
+import com.ib.certificate.exception.BadExpirationDateException;
+import com.ib.certificate.exception.InvalidCertificateTypeException;
+import com.ib.certificate.exception.IssuerUnauthorizedException;
+import com.ib.user.User.Role;
+
+@Service
+public class CertificateService implements ICertificateService {
+	@Autowired
+	private ICertificateRepo certificateRepo;
+	@Autowired
+	private ICertificateRequestService certificateRequestService;
+	
+	@Override
+	public Certificate findById(Long id) {
+		return certificateRepo.findById(id).orElse(null);
+	}
+
+	@Override
+	public Certificate accept(CertificateRequest req) {
+		certificateRequestService.setStatus(req, Status.ACCEPTED);
+		
+		Certificate c = new Certificate();
+		c.setIssuer(req.getIssuer());
+		c.setParent(req.getParent());
+		c.setType(req.getType());
+		c.setValidFrom(LocalDateTime.now());
+		c.setValidTo(req.getValidTo());
```

Comment:
Give `Certificate` a constructor that takes a `CertificateRequest`

Reply:
Done in [dc5447c](https://github.com/magley/sssscs/pull/3/commits/dc5447cee3548a166981e44c59f23f3fd00b2b66).